File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s010698290_commented.py
n, m = map(int, input().split())

solved = [False] * (n + 1)
penal = [0] * (n + 1)

for _ in range(m):
    parts = input().split()
    num = int(parts[0])
    action = parts[1]
    if action[0] == 'A':
        solved[num] = True
    else:
        if not solved[num]:
            penal[num] += 1

correct_ans = 0
num_penal = 0
for i in range(1, n + 1):
    if solved[i]:
        correct_ans += 1
        num_penal += penal[i]

print(correct_ans, num_penal)
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C/s010698290_commented.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    int n, m;
    scanf("%d %d", &n, &m);

    int *penal = (int *)malloc((n + 1) * sizeof(int));
    char *solved = (char *)malloc((n + 1) * sizeof(char));

    memset(solved, 0, (n + 1) * sizeof(char));
    memset(penal, 0, (n + 1) * sizeof(int));

    for (int i = 0; i < m; i++) {
        int num;
        char str[2];
        scanf("%d %s", &num, str);
        if (str[0] == 'A') {
            solved[num] = 1;
        } else {
            if (!solved[num]) {
                penal[num]++;
            }
        }
    }

    int correct_ans = 0, num_penal = 0;
    for (int i = 1; i <= n; i++) {
        if (solved[i]) {
            correct_ans++;
            num_penal += penal[i];
        }
    }

    printf("%d %d\n", correct_ans, num_penal);

    free(penal);
    free(solved);

    return 0;
}
// End of Code
